README.md

```diff
@@ -161,8 +178,9 @@ Unit's output should contain both snippets, neatly organized:
 }
 ```
 
-For full details of configuration management, see the
-[docs](https://unit.nginx.org/configuration/#configuration-management).
+## WebAssembly
+Unit supports deploying WASM as a supported application.
```

Comment:
1. Above we talk about "running apps" but here we're talking about "deploying".
2. I'm not sure that calling out WebAssembly is warranted over all the other runtimes. I do like the idea of promoting it but retain a small reservation.
3. The abbreviation is Wasm so we should either avoid using two terms or introduce the abbreviation if we intend to use it again.
```suggestion
Unit supports running WebAssembly (Wasm) Components.
```

Reply:
About your second point:
I did deliberately call out WebAssembly where other languages are not. My rationale is articulated in the merge request description. My goal is not to promote WebAssembly, but to solve a discovery issue that we have.

Coming to the project, I encountered the unit-wasm repo first. I did not know it was deprecated. I was confused as to why it wasn't mentioned in our Readme since it is a separate repository. I did not realize we had a more up to date module in the Unit repository until someone told me.

Most people are going to find us through Github before they look at our docs site. I am willing to hazard a bet that people will find unit-wasm before they find our largely undocumented WASI module.

Perhaps a better solution is to update unit-wasm with a deprecation notice that redirects to whatever documentation we do have for our WASI module.

Additionally, I think it might be worth explicitly listing all of the languages we support (and probably separate quick starts for each) in the Readme; lest someone skimming our readme miss the one single line we have saying that we support multiple languages. They would come off thinking we only supported PHP.

Happy to either leave this here, or to go update unit-wasm. Please let me know which is preferrable.

Reply:
I _am_ hesitant to explicitly mark the `wasm` language module as deprecated at the moment as it handles some cases that the new one doesn't. E.g ingress/egress of > 4GiB data... we also don't get the same amount of request information, well, not that I found in the C API anyway, see <https://gist.github.com/ac000/906707af37b42cdcc9d3cc4c2715da99#file-000-wasi-http-txt-L21>

Reply:
Your observations are spot-on.

IMO, the job of the readme is to let people know if this is something they should pay attention to, and if so, to get them up and running very quickly. This PR is compatible with that :)

Deprecating unit-wasm and libunitwasm are equally important to that goal.

Reply:
> Deprecating unit-wasm and libunitwasm are equally important to that goal.

See above.

I _could_ add a note to the unit-wasm README that points people to `wasm-wasi-component` _if_ they are happy with its caveats (and yes that includes requiring _all_ the rust stuff, or running in memory constrained environments)...

Reply:
About the memory constraints.

When uploading a file and simply writing it to disk...

>Yes, this does mean that the wasm-wasi-component application process does end   
up consuming quite a bit of RAM. When trying to upload the 2GiB file, I clock   
it using about ~4GiB, though this does get freed afterwards.                    
                                                                                
> For comparison, the C language module consumes a steady ~70MiB.

This _may_ be fixable...

Reply:
> I could add a note to the unit-wasm README that points people to wasm-wasi-component if they are happy with its caveats (and yes that includes requiring all the rust stuff, or running in memory constrained environments)...

I think a note would be excellent. I am happy to add one myself. I don't think it's worth unsubstantiated fearmongering about Rust in memory constrained environments in that comment. That would only either confuse new users or potentially turn off people who are using Rust.

Also, in such environments would a user really be deploying a WebAssembly component?

Reply:
It sounds like these performance issues should be an issue, bug, or some other tracked work we can prioritize though.